src/aap_eda/wsapi/consumers.py

```diff
@@ -498,20 +497,32 @@ def _get_url(self, message: ActionMessage, inputs: dict) -> str:
     @database_sync_to_async
     def get_file_contents_from_credentials(
         self, activation: models.Activation
-    ) -> tp.Optional[dict]:
-        file_templates = {}
+    ) -> tp.Optional[list[FileContentMessage]]:
+        file_template_names = []
+        file_messages = []
         for eda_credential in activation.eda_credentials.all():
             inputs = yaml.safe_load(eda_credential.inputs.get_secret_value())
             injectors = eda_credential.credential_type.injectors
+            binary_fields = []
+            for field in eda_credential.credential_type.inputs.get(
+                "fields", []
+            ):
+                if field.get("format") in BINARY_FORMATS:
+                    binary_fields.append(field["id"])
```

Comment:
Is `id` a required field? it may be better to use `field.get("id")` to avoid potential KeyError

Reply:
@hsong-rh In a Credential Schema a id is a required field. We can have an empty schema but not a field with a missing id.